### PR TITLE
✨ Let value_map take a regex pattern also

### LINKED ISSUE #371: let value_map take a regex pattern not in a dict
m=r"asdfasdfsadf" will be treated like {m: lambda x: x}

### PR DESCRIPTION
**Pull request by fiendish**
closes #371 

¿What happens if I also say that it...
closes #372 ?

### CODE REVIEW

**Comment by fiendish** [comment]
What happens if I also say that this closes #372

File: kf_lib_data_ingest/etl/extract/operations.py [hunk 1]
```diff
@@ -51,30 +51,40 @@ def value_map(m, in_col, out_col):
     Wraps the value mapping operation in a function that takes a DataFrame as
     input and returns a single column DataFrame (plus index).
 
-    :param m: Can be a dict or a function that takes a basic string. If a
-        function, the return columns values wil be the result of applying that
-        function to each cell of the <in_col> column in the DataFrame. If a
-        dict, each dict key is a regex pattern that is forced to have ^$
+    :param m: Can be a dict, regex string, or a function that takes a string.
```
Comment (znatty22) [inline kf_lib_data_ingest/etl/extract/operations.py]:
```suggestion
    :param m: Can be a dict, regex pattern string, or a function that takes a string.
```

File: kf_lib_data_ingest/etl/extract/operations.py [hunk 2]
```diff
@@ -51,30 +51,40 @@ def value_map(m, in_col, out_col):
...
+
+        If a function, the return column values wil be the result of applying
+        that function to each cell of the <in_col> column in the DataFrame.
+
+        If a dict, each dict key is a regex pattern that is forced to have ^$
```
Comment (znatty22) [inline kf_lib_data_ingest/etl/extract/operations.py]:
```suggestion
        If a dict, each dict key is a regex pattern string that is forced to have ^$
```

File: kf_lib_data_ingest/etl/extract/operations.py [hunk 3]
```diff
@@ -51,30 +51,40 @@ def value_map(m, in_col, out_col):
...
         be either replaced by the indicated constant value or they will be
         filled by application of the indicated function to the value of the
         cell in <in_col>. If the regex pattern has capture groups, those become
-        the function inputs instead of the entire cell value. For more details,
-        read the docstring for safe_pandas_replace in common/pandas_utils.py
+        the function inputs instead of the entire cell value.
+        For more details, read the docstring for safe_pandas_replace in
+        common/pandas_utils.py
+
+        If a regex string, the result will be the asme as if it had been
```
Comment (znatty22) [inline kf_lib_data_ingest/etl/extract/operations.py]:
```suggestion
        If a regex pattern string, the result will be the same as if it had been
```

**Pull request merged**